puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/infra/updateJobs.groovy

```diff
@@ -0,0 +1,54 @@
+pipeline {
+    agent { label 'fast' }
+
+    stages {
+        stage('Setup workspace') {
+            steps {
+                checkout([
+                    $class : 'GitSCM',
+                    branches : [[name: 'master']],
+                    extensions: [[$class: 'CleanCheckout']],
+                    userRemoteConfigs: [
+                        [url: 'https://github.com/theforeman/foreman-infra.git']
+                    ]
+                ])
+
+                dir("./ci") {
+                    sh """
+                    virtualenv update_jobs
+                    source update_jobs/bin/activate
+                    pip install jenkins-job-builder
+                    deactivate
+                    """
+                }
+              }
+        }
+
+        stage('Update ci.centos.org jobs') {
+            steps {
+                dir("./ci") {
+                    withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: 'jenkins-centos', passwordVariable: 'PASSWORD', usernameVariable: 'USERNAME']]) {
+                        sh """
+                        source update_jobs/bin/activate
+                        mv ./centos.org/centos_jenkins.ini.example ./centos.org/jenkins_jobs.ini
+                        jenkins-jobs --conf ./centos.org/jenkins_jobs.ini --user ${env.USERNAME} --password ${env.PASSWORD} update --delete-old -r ./centos.org
```

Comment:
this will include stuff from `centos.org/ansible` and die in flames. you want to only scan `centos.org/jobs` for jobs :)
also, for the relative paths in the job definitions to work, you need to execute this from the `centos.org` folder

Reply:
I tested it locally with the test command and nothing broke, but it's still probably a better practice.

Reply:
interesting, test is what exploded for me. but hey, now it's the way it did not explode ;) thanks